### PR TITLE
Improve the performance of bayesian_blocks (binned or unbinned events)

### PR DESCRIPTION
**Pull request by Gabriel-p**
### Description

A very small change to improve the performance of `bayesian_blocks()`. In my tests it runs ~1.4x faster, between 30-40% of time saved.

```
In [2]: %timeit -n 5 -r 5 bayesian_block(x)
10.7 s ± 39.9 ms per loop (mean ± std. dev. of 5 runs, 5 loops each)

In [3]: %timeit -n 5 -r 5 bayesian_block_new(x)
6.34 s ± 668 ms per loop (mean ± std. dev. of 5 runs, 5 loops each)
```

Could somebody confirm this?

### CODE REVIEW

File: CHANGES.rst
Comment (pllim) [inline CHANGES.rst:78]: Not sure if this belongs in "new feature," but this isn't a bug fix or API change either, so maybe under whatever we have for "others"?

```suggestion
- Improve performance of ``bayesian_blocks()`` by removing one ``no.log()`` call. [#11356]
```

**Review by pllim — COMMENTED** [review]
Good catch!

```python
In [2]: import numpy as np

In [3]: a = np.array([1000] * 10)

In [4]: b = np.array([100000] * 10)

In [5]: %timeit np.log(b) - np.log(a)
2.26 µs ± 11.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [6]: %timeit np.log(b / a)
1.63 µs ± 2.52 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

I'll keep this open for a bit more so subpackage maintainers can review but this LGTM. If you don't hear back in 1-2 weeks, feel free to ping me again and I'll just approve it after the change log entry is taken care of.

File: CHANGES.rst
Comment (Gabriel-p) [reply CHANGES.rst:78]:
```suggestion
- Improve performance of `bayesian_blocks()` by removing one `np.log()` call
```

**Review by larrybradley — CHANGES_REQUESTED** [review]
Thanks, @Gabriel-p!  LGTM.  Can you please move the change log entry the "Other Changes and Additions" section, since this really isn't a new feature?

**Comment by Gabriel-p** [comment]
Sorry, I did not notice that I was adding the change to that section. How does it look now @larrybradley?

File: CHANGES.rst
Comment (larrybradley) [inline CHANGES.rst:78]: Can you remove this extra line?

Comment (Gabriel-p) [reply CHANGES.rst]: Done

**Comment by larrybradley** [comment]
Thanks, @Gabriel-p.  This looks great!  I have one more small request before merging -- since this is a single-line small code change, can you please squash these commits?

**Comment by Gabriel-p** [comment]
How's that?

**Review by pllim — APPROVED** [review]
Perfect. Thanks!

**Review by larrybradley — APPROVED** [review]
Thanks, @Gabriel-p!

**Comment by pllim** [comment]
Remote data job failure is unrelated, so merging. Thanks, everyone!

**Pull request merged**
